spec/factories/factories.rb

```diff
@@ -8,11 +8,24 @@
     category    { Faker::Lorem.words(4).join }
     description { Faker::Lorem.characters(255) }
     attachment  { Faker::File.file_name('path/to/') }
+    association :user, factory: :user

```

Comment:
attachment is a real column in DB?
change factory to sample as I mentioned before

`    file { Rack::Test::UploadedFile.new(Rails.root.join('spec', 'files', 'test.pdf')) }`

puts file in some test folder as here I made in spec/files/test.pdf you can do similar way